libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -140,36 +142,45 @@ void IRGeneratorForStatements::initializeStateVar(VariableDeclaration const& _va
 	if (_varDecl.value())
 	{
 		_varDecl.value()->accept(*this);
-		string value = m_context.newYulVariable();
-		Type const& varType = *_varDecl.type();
-
-		m_code << "let " << value << " := " << expressionAsType(*_varDecl.value(), varType) << "\n";
-		m_code << IRStorageItem{m_context, _varDecl}.storeValue(value, varType);
+		writeToLValue(IRLValue{
+			*_varDecl.annotation().type,
+			IRLValue::Storage{
+				util::toCompactHexWithPrefix(m_context.storageLocationOfVariable(_varDecl).first),
+				m_context.storageLocationOfVariable(_varDecl).second
+			}
+		}, *_varDecl.value());
+		m_currentLValue.reset();
```

Comment:
Why is this needed and why is it not part of `writeToLValue`?

Reply:
Well ``writeToLValue`` now gets the lvalue it writes to as argument now and has no dealings with ``m_currentLValue``...

Reply:
Ok if you don't access `m_currentLValue`, why do you reset it?